### PR TITLE
Updates the documentation on versioning

### PR DESCRIPTION
**Pull request by joeldudleyr3**

### CODE REVIEW

File: docs/source/upgrading-cordapps.rst [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+This document will consider the following types of versioning:
+
+* Flow versioning
+* State and contract versioning
+* State and state schema versioning
+* Serialisation of custom types
+
+Flow versioning
+---------------
+Any flow that initiaties other flows must be annotated with the ``@InitiatingFlow`` annotation, which is defined as:
```
Comment (CaisR3) [inline docs/source/upgrading-cordapps.rst]: spelling - initiates

File: docs/source/upgrading-cordapps.rst [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+Assumptions
+^^^^^^^^^^^
+
+* All nodes in the business network can be shut down for a period of time
+* All nodes retire the old flows and adopt the new flows at the same time
+
+Process
+^^^^^^^
+
+1. Update the flow and test the changes. Uncrement the flow version number in the ``InitiatingFlow`` annotation
```
Comment (CaisR3) [inline docs/source/upgrading-cordapps.rst]: Spelling - increment

File: docs/source/upgrading-cordapps.rst [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+        }
+    }
+
+This code shows a flow that in its first version expected to receive an Int, but in subsequent versions was modified to
+expect a String. This flow is still able to communicate with parties that are running the older CorDapp containing
+the older flow.
+
+How do I deal with interface changes to inlined sub flows?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Here is an example of an in-lined subflow:
```
Comment (CaisR3) [inline docs/source/upgrading-cordapps.rst]: we sometimes use subflow, sub-flow and sub flow - sorry to be picky, should we stick to one?

File: docs/source/upgrading-cordapps.rst [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+
+Suspended flows
+^^^^^^^^^^^^^^^
+Currently, serialised flow state machines persisted in the node's database cannot be updated. All flows must finish
+before the updated flow classes are added to the node's plugins folder.
+
+Flows that don't create sessions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Flows which are not an ``InitiatingFlow`` or ``InitiatedBy`` flow, or inlined subflows that are not called from an
+``InitiatingFlow`` or ``InitiatedBy`` flow , can be updated without consideration of backwards-compatibility. Flows of
```
Comment (CaisR3) [inline docs/source/upgrading-cordapps.rst]: extra space "flow ,"

File: docs/source/upgrading-cordapps.rst [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+* If a node has not yet run the contract upgrade authorisation flow, they will not be able to upgrade the contract
+  and/or state objects
+* Upgrade authorisations can subsequently be deauthorised
+* Upgrades do not have to happen immediately. For a period, the two parties can use the old states and contracts
+  side-by-side
+* State schema changes are handled separately
+
+Writing new states and contracts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* If a property is removed from a state, any references of it must be removed from the contract code. Otherwise, you
```
Comment (CaisR3) [inline docs/source/upgrading-cordapps.rst]: "to it"? instead of "of it"?

File: docs/source/upgrading-cordapps.rst [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+  * This will change when the contract code as an attachment feature has been fully implemented.
+
+Permissioning
+^^^^^^^^^^^^^
+* Only node administrators are able to run the contract upgrade authorisation and deauthorisation flows
+
+Logistics
+^^^^^^^^^
+* All nodes need to run the contract upgrade authorisation flow
+* Only one node should to run the contract upgrade initiation flow. If multiple nodes run it for the same ``StateRef``,
```
Comment (CaisR3) [inline docs/source/upgrading-cordapps.rst]: extra "to" in "should to run"

**Comment by joeldudleyr3** [comment]
Addressed the review comments.

**Pull request merged**